src/nvim/os/input.c

```diff
@@ -306,37 +306,35 @@ static uint8_t check_multiclick(int code, int grid, int row, int col)
   static int orig_mouse_row = 0;
   static uint64_t orig_mouse_time = 0;  // time of previous mouse click
 
-  if (code == KE_LEFTRELEASE
-      || code == KE_RIGHTRELEASE
-      || code == KE_MIDDLERELEASE
-      || code == KE_MOUSEDOWN
-      || code == KE_MOUSEUP
-      || code == KE_MOUSELEFT
-      || code == KE_MOUSERIGHT
-      || code == KE_MOUSEMOVE) {
+  if ((code >= KE_MOUSEDOWN && code <= KE_MOUSERIGHT) || code == KE_MOUSEMOVE) {
```

Comment:
Vim also keeps the number of clicks for mouse release events, so using the number of clicks below instead of returning 0 here matches Vim's behavior better.